THREADING.md

```diff
@@ -100,6 +104,13 @@ direct connection to libgcrypt and thus has not convenience functions for
 it (but libgcrypt has macros). Read libgcrypt's
 [threading documentation for more information](http://www.gnupg.org/documentation/manuals/gcrypt/Multi_002dThreading.html)
 
+libssh2 provides a `libssh2_init` function, which is called
+automatically by libssh2 code if needed.  However, `libssh2_init`
+is not thread-safe, and the way in which it is automatically called
+is also not thread-safe, so in a multi-threaded environment you may need
+to call libssh2_init directly to prevent the thread-unsafe automatic
+initialization.
```

Comment:
I guess I was treating this as related to thread safety, and staying consistent with "we don't set up thread safety for dependencies" :-)

But yes, we could call it as part of `libgit2_init` and that would probably be for the best.  libssh2 keeps an internal init count so it shouldn't hurt.  I can do that in a separate PR - I pulled out this docs section.